### ISSUE TITLE
UE5.0 Source Code build cannot find plugin

### ISSUE THREAD

**Issue by ShinraCorpMixer** [PlayFab/gsdk#124]
**Describe the bug**
When following the documentation to install the GSDK for Unreal Engine, the Unreal Editor (version 5.0 built from the release branch source code on GitHub) cannot "find" the plugin. 

However, if the plugin is disabled, the editor launches and can find the plugin in the plugins window. The moment you enable it and restart the editor. It will say that it cannot find the plugin anymore again.

https://docs.microsoft.com/en-us/gaming/playfab/features/multiplayer/servers/server-sdks/unreal-gsdk/third-person-mp-example-gsdk-project-setup

**To Reproduce**
1.  Complete the prerequisites that the document asks for (Unreal Engine source code build, have a Dedicated Server version of the game, have the game work with networking.)
2. Create the Plugins folder in the Game's Root directory, then create a folder in the Plugins folder called PlayFabGSDK, copy and paste all of the files found in the {depot}\GSDK\gsdk\UnrealPlugin folder and place it in the PlayFabGSDK folder.
3. Add the plugin in the .uproject file via text editor.
4. Add the plugin in the modules ([ProjectName].Build.cs)
5. Right click on the .uproject file and click on Generate Visual Studio files.
6. Build the project.
7. Attempt to open the project with Unreal Editor.

**Expected behavior**
The Unreal Editor launches with the plugin enabled.

**Screenshots**
![image](https://user-images.githubusercontent.com/55590236/162451767-66620b1d-23bb-43bc-b634-3e7afd3a2cc1.png)

![image](https://user-images.githubusercontent.com/55590236/162451944-5abf7193-7b80-4dd0-8a62-9dab91e829f6.png)

![image](https://user-images.githubusercontent.com/55590236/162452257-c9edda3e-f3eb-4197-8a80-7585da8d89af.png)

![image](https://user-images.githubusercontent.com/55590236/162472222-8db99287-f0e6-4581-a938-04ee3bb754fa.png)

**Desktop (please complete the following information):**
 - OS: Windows 11 Pro
 - Version 22H2

**Additional context**
I have tried reversing the array entry in the .uproject file by having enabled: true first and then the name of the plugin. Didn't work.

**Comment by nassosterz-ms** [comment]
Are you using the official release of Unreal Engine 5 or the early access version?

**Comment by ShinraCorpMixer** [comment]
> Are you using the official release of Unreal Engine 5 or the early access version?

The official one that came out this week :) And no worries! Let me know if you need more information.

![image](https://user-images.githubusercontent.com/55590236/162469568-6823f793-e06b-46a4-ba4d-054dd873f703.png)

**Comment by nassosterz-ms** [comment]
One thing I notice is a discrepancy in your file explorer screenshot.

Here is mine (this works for UE5 - early access).

![fe_gsdk](https://user-images.githubusercontent.com/93945007/162473342-f7d52ee2-09de-4692-bc5e-7bb11be2f9ec.PNG)

**Comment by ShinraCorpMixer** [comment]
Yeah I noticed that as well, I might have created it by accident when packaging the plugin in the Unreal Editor. So I deleted everything in the folder and readded back from the depot. 

Same error sadly... it sees the .uplugin but doesn't want to work for some reason....

![image](https://user-images.githubusercontent.com/55590236/162474862-20b6b5fa-44dd-4b4b-b09b-c53e1c206c5a.png)

**Comment by nassosterz-ms** [comment]
First of all, I was able to replicate that. Saw exactly the same things as what you reported here.

I was able to resolve that after I tried many things to be fair. What I am describing below occurs after I have completed steps 1-4 as you outline them in your description.

Looks like UE5 release has an issue with the directory structure of UnrealBuildTool. In my case, I couldn't Generate Visual Studio Project Files. It is not clear from your description whether you were succeeding in that, in my case it was failing.

The fix I found for that was to create a .bat file in Path to UE5UE5Release-5.0EngineBinariesDotNETUnrealBuildTool with the following content

**_UnrealBuildTool.exe -projectfiles -project=path to game projectgameproject.uproject -game -engine -rocket -progress_**

I ran that, then opened Visual Studio and built the game for Development Editor - Win64. I got a compilation error due to using a variable name that is deprecated in UE5 (PR coming soon).

After I fixed that, compilation succeeded. I was able then to open the game from .uproject without getting that error.
